op-batcher/batcher/driver_test.go

```diff
@@ -0,0 +1,79 @@
+package batcher
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/testlog"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr/mocks"
+)
+
+// TestSendTransaction tests that the driver can send a transaction
+// through the txmgr.
+func TestSendTransaction(t *testing.T) {
```

Comment:
I'm not sure what this test gain us. We check the ability of "can send" with types + in e2e tests. There are two other types of bug which would cause a failure to send: A misconfigured transaction manger which this test will not catch & not actually calling sendTransaction in the batcher. This does ensure that we call `sendTransaction`, but it's a very obvious function that needs to be called with many other layers of testing.

Reply:
IMO - since I touched an external function, it'd be strange not to have any tests. Although it _may_ be tested elsewhere, there should be a _local_ test - and although yes it's just testing the happy path, that's the bare minimum.